xc/device_common.ml

```diff
@@ -266,6 +270,25 @@ let list_frontends ~xs ?for_devids domid =
                                ) devids)
                  ) kinds)
 
+(* Network SR-IOV VF device *)
+let make_nsv_device ?(back_domid=0) front_domid devid =
+  let frontend = { domid = front_domid; kind = NetSriovVf; devid = devid } in
+  let backend = { domid = back_domid; kind = NetSriovVf; devid = devid } in
+  { backend = backend; frontend = frontend }
+
+let list_xenserver_devices ~xs ?for_devids domid =
+  let xenserver_dir = sprintf "/local/domain/%d/xenserver/device" domid in
+  let mk = function
+    | NetSriovVf ->
+      let dir = sprintf "%s/%s" xenserver_dir (string_of_kind NetSriovVf) in
+      let devids = filter_on_devids ~xs ?for_devids dir in
+      Some (List.map (make_nsv_device domid) devids)
```

Comment:
Here the `back_domid` has to be always `0`?

Reply:
For network SR-IOV, I think it is true.
@robhoes  May I have your opinion on this?